Mathlib/RingTheory/DedekindDomain/FiniteAdeleRing.lean

```diff
@@ -274,20 +310,18 @@ open ProdAdicCompletions.IsFiniteAdele
 
 /-- The finite adèle ring of `R` is the restricted product over all maximal ideals `v` of `R`
 of `adicCompletion` with respect to `adicCompletionIntegers`. -/
-noncomputable def finiteAdeleRing : Subring (K_hat R K) where
-  carrier := {x : K_hat R K | x.IsFiniteAdele}
-  mul_mem' hx hy := mul hx hy
-  one_mem' := one
-  add_mem' hx hy := add hx hy
-  zero_mem' := zero
-  neg_mem' hx := neg hx
-#align dedekind_domain.finite_adele_ring DedekindDomain.finiteAdeleRing
-
-variable {R K}
-
-@[simp]
-theorem mem_finiteAdeleRing_iff (x : K_hat R K) : x ∈ finiteAdeleRing R K ↔ x.IsFiniteAdele :=
-  Iff.rfl
-#align dedekind_domain.mem_finite_adele_ring_iff DedekindDomain.mem_finiteAdeleRing_iff
+def FiniteAdeleRing : Type _ := (
+  { carrier := {x : K_hat R K | x.IsFiniteAdele}
+    mul_mem' := mul
+    one_mem' := one
+    add_mem' := add
+    zero_mem' := zero
+    algebraMap_mem' := algebraMap'
+  } : Subalgebra K (K_hat R K))
+#align dedekind_domain.finite_adele_ring DedekindDomain.FiniteAdeleRing
+
+instance : CommRing (FiniteAdeleRing R K) := Subalgebra.toCommRing _
+
```

Comment:
Why do you want to get rid of the lemma `mem_finiteAdeleRing_iff`?

Reply:
I've changed `FiniteAdeleRing` from a term to a type, so the lemma no longer makes any sense as it stands, and I couldn't think of what to replace it with.

Reply:
Coe lemma added!

Reply:
Thanks, definitely LGTM! IMHO this PR is ready to go.